### PR TITLE
interfaces/builtin: fix pulseaudio apparmor rules

### PR DESCRIPTION
**Pull request by bergotorino**
The pulseaudio snap run on the kvm revealed that the pactl tries to
access several locations which were not described by the AppArmor
rules. This commit makes them accessible so that the pactl tool
can be executed.

In particular the locations in question were:
 - /run/user/${USERID}
 - /run/user/${USERID}/pulse
 - /usr/share/applications

### CODE REVIEW

**Comment by jhodapp** [comment]
Looks good to me as well although the autopkgtest needs looking at.

**Comment by mvo5** [comment]
Looks good, please ignore the trusty autopkgtest failure. This needs a review from security (@jdstrand) though.

File: interfaces/builtin/pulseaudio.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ const pulseaudioConnectedPlugAppArmor = `
 
 owner /{,var/}run/pulse/ r,
 owner /{,var/}run/pulse/native rwk,
+owner /run/user/[0-9]*/{,pulse/} rw,
```
Comment (jdstrand) [inline interfaces/builtin/pulseaudio.go]: This rule isn't quite right. You want this instead:
```
owner /run/user/[0-9]*/ r,
owner /run/user/[0-9]*/pulse/ rw,
```

A snap that 'plugs: [ pulseaudio ]' should not be able to manipulate '/run/user/[0-9]*/'. Something else should be creating '/run/user/[0-9]*/' (in fact, snap-confine will when is supports XDG_RUNTIME_DIR, which is pending).

File: interfaces/builtin/pulseaudio.go [hunk 2]
```diff
@@ -92,6 +93,11 @@ owner /{,var/}run/pulse/** rwk,
 
 # Shared memory based communication with clients
 /{run,dev}/shm/pulse-shm-* rwk,
+
+/usr/share/applications/ r,
+
+owner /run/pulse/native/ rwk,
+owner /run/user/[0-9]*/{,pulse/} rw,
```
Comment (jdstrand) [inline interfaces/builtin/pulseaudio.go]: Same here, please use:
```
owner /run/user/[0-9]*/ r,
owner /run/user/[0-9]*/pulse/ rw,
```

**Pull request merged**